### ISSUE TITLE
NaNs in functionValues after parameter update

### ISSUE THREAD

**Issue by jian16** [Microsoft/CNTK#521]
I am training a model and after a few epochs, the loss function dropped significantly, which is not expected. After debugging into the code, I found NaN values in BIAS LearnableParameter. Any idea? How could this happen? Is there a good way to debug? I just turned on TRACK_GAP_NANS expecting to print more information.

Thanks,
Jian
